keps/sig-instrumentation/3466-kubernetes-component-health-slis/README.md

```diff
@@ -0,0 +1,504 @@
+definition](https://kubernetes.io/docs/concepts/overview/kubernetes-api/#api-versioning)
+or by redefining what graduation means.
+
+In general we try to use the same stages (alpha, beta, GA), regardless of how the
+functionality is accessed.
+
+[feature gate]: https://git.k8s.io/community/contributors/devel/sig-architecture/feature-gates.md
+[maturity-levels]: https://git.k8s.io/community/contributors/devel/sig-architecture/api_changes.md#alpha-beta-and-stable-versions
+[deprecation-policy]: https://kubernetes.io/docs/reference/using-api/deprecation-policy/
+
+Below are some examples to consider, in addition to the aforementioned [maturity levels][maturity-levels].
+-->
+
+#### Alpha
+
+- Feature implemented behind a feature flag
+- Initial e2e tests completed and enabled
+- Feature implemented for apiserver
+
+#### Beta
+
+- Gather feedback from developers 
+- Feature implemented for other Kubernetes Components
+
+#### GA
+
+- Several cycles of bake-time
+- Graduation of metrics to stable status
+
+
+#### Deprecation
+
+- Announce deprecation and support policy of the existing flag
+- Two versions passed since introducing the functionality that deprecates the flag (to address version skew)
+- Address feedback on usage/changed behavior, provided on GitHub issues
+- Deprecate the flag
+
+### Upgrade / Downgrade Strategy
+
+This is a new metrics endpoint and should not affect upgrade/downgrade strategy with the exception that
+if you are scraping this endpoint, downgrading may remove this endpoint from the Kubernetes components
+and you may end up missing these metrics.
+
+### Version Skew Strategy
+
+We do not plan to modify these metrics, so it should be safe for version skew. 
+
+## Production Readiness Review Questionnaire
+
+
+### Feature Enablement and Rollback
+
+We will target this feature behind a flag `ComponentHealthSLIsFeatureGate`
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+
+- [ ] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name: `ComponentHealthSLIsFeatureGate`
+  - Components depending on the feature gate:
+    + apiserver
+    + kubelet
+    + scheduler
+    + controller-manager
+    + kube-proxy
+
+
+###### Does enabling the feature change any default behavior?
+
+Yes it will expose a new metrics endpoint.
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+Yes. But it will remove the metrics endpoint
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+It will expose the metrics endpoint again
+
+###### Are there any tests for feature enablement/disablement?
+
+We intend to add them with our e2e tests.
+
+### Rollout, Upgrade and Rollback Planning
+
+I'm not sure these are super relevant for this feature.
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+This feature should not cause rollout failures. If it does, we can disable the feature.
```

Comment:
I agree it can't impact already running workloads.

In a pathological case I can imagine this feature, e.g. crashing a component so it can potentially break the rollout (although fairly unlikely that such thing won't be caught by tests).

Reply:
Added some text to that effect.